jaxrs/WFLY-13122_EAP7-1494_Add_Json_Merge_Patch_support.adoc

```diff
@@ -0,0 +1,146 @@
+---
+categories:
+  - jaxrs
+stability-level: default
+issue: https://github.com/wildfly/wildfly-proposals/issues/664
+feature-team:
+ developer: ema
+ sme:
+  - jamezp
+  - ronsigal
+  - liweinan
+ outside-perspective:
+  - honza-kasik
+promotes:
+promoted-by:
+---
+= Add JSON Merge Patch support
+:author:            Jim Ma
+:email:             ema@redhat.com
+:toc:               left
+:icons:             font
+:keywords:          json merge patch,resteasy,jaxrs
+:idprefix:
+:idseparator:       -
+:issue-base-url:    https://issues.jboss.org/browse
+
+== Overview
+This feature is to support JSON Merge Patch spec as described in: https://tools.ietf.org/html/rfc7386
+With JSON merge patch support, the client can send a JSON http request with Content-Type "application/merge-patch+json",
+and the JSON merge request will be directly apply to the target resource with the merge function as spec described.
+
+RESTEasy provides two implementation for JSON merge patch. One based on jsonp(abbreviation jsonp will be used in this page) and another one is based on
+json-patch and jackson2(abbreviation jackson2 will be used in this page). By default, RESTEasy uses jsonp implementation, while the jackson2 implementation is only enabled
+when the context parameter `resteasy.patchfilter.legacy=true` is set.
+
+The JSON merge patch functionality in RESTEasy can be completely disabled by setting the context parameter `resteasy.patchfilter.disabled=true`.
+When this parameter is enabled, any JSON merge patch requests will be ignored.
+
+=== User Stories
+
+When user want to merge a JSON object to the target JSON object by specifying the merge part the user wants to make, instead of providing the full updated version of the json content.
+
+== Issue Metadata
+
+=== Issue
+
+* {issue-base-url}/EAP7-1494 [EAP7-1494]
+* {issue-base-url}/WFLY-13412 [WFLY-13412]
+
+=== Related Issues
+
+* {issue-base-url}/RESTEASY-2567 [RESTEASY-2567]
+* {issue-base-url}/WFLY-19981 [WFLY-19981]
+* {issue-base-url}/WFLY-19833 [WFLY-19833]
+
+=== Affected Projects or Components
+
+- https://github.com/resteasy/resteasy[resteasy]
+
+=== Other Interested Projects
+
+=== Relevant Installation Types
+
+* Traditional standalone server (unzipped or provisioned by Galleon)
+* OpenShift Source-to-Image (S2I)
+* Bootable jar
+
+== Requirements
+
+=== Hard Requirements
+
+* RESTEasy includes this feature support and is installed in WFLY
+* The PatchMethodFilter is enabled by default. It can be disabled by setting the resteasy.patchfilter.disabled configuration to true.
+* The `resteasy-patchfilter-disabled` configuration will be added to jaxrs subsystem, but this configuration is optional.
+The jaxrs subsystem version will be upgraded from 3.0 to 3.1 for this change.
```

Comment:
We don't need to be specific with the versions here, but if we do it will be 4.0.